nrf52-hal-common/src/uarte.rs

```diff
@@ -116,11 +124,20 @@ impl<T> Uarte<T> where T: Instance {
             return Err(Error::TxBufferTooLong);
         }
 
+        // We can only DMA out of RAM
+        if !crate::slice_in_ram(tx_buffer) {
+            return Err(Error::BufferNotInRAM);
+        }
```

Comment:
@simonsso You spoke out against this approach in #37. I still think returning the error is clearly the best approach and would like to merge this as-is, but didn't want to sneak this by you.